test/reader.js

```diff
@@ -0,0 +1,328 @@
+/* eslint-env mocha */
+'use strict'
+
+const reader = require('./../src').reader
+
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+const expect = chai.expect
+const IPFS = require('ipfs')
+const path = require('path')
+const os = require('os')
+const pull = require('pull-stream')
+const collect = require('pull-stream/sinks/collect')
+
+module.exports = (repo) => {
+  describe('reader', function () {
+    this.timeout(30 * 1000)
+
+    let ipfs
+
+    before((done) => {
+      ipfs = new IPFS({
+        repo: path.join(os.tmpdir(), 'unixfs-test-' + Math.random())
+      })
+      ipfs.on('ready', () => {
+        done()
+      })
+      ipfs.on('error', (error) => {
+        done(error)
+      })
+    })
+
+    after((done) => {
+      ipfs.stop(done)
+    })
+
+    it('fails on invalid path', (done) => {
+      pull(
+        reader('?!?', ipfs._ipldResolver),
+        collect((error) => {
+          expect(error.message).to.contain('Non-base58 character')
```

Comment:
This error should also mention the value that was invalid, so the user can easily see what the wrong input is.

Reply:
Done!

Reply:
This error message comes from [base-x](https://github.com/cryptocoinjs/base-x/blob/master/index.js#L84) so it could be improved there but it appears it's [depended on](https://github.com/ipfs/js-ipfs/blob/master/src/http/gateway/resources/gateway.js#L65) in a few places so might be a more disruptive job than it appears.